pkg/deps/annotation_matcher.go

```diff
@@ -75,9 +81,12 @@ func AnnotationMatcher(criteria *AnnotationCriteria) Matcher {
 		}
 
 		matchesAnnot := func(key string, vals []string, annots map[string]string) bool {
+			if _, ok := annots[key]; ok && len(vals) == 0 {
+				return true
```

Comment:
It's not wrong, but it's surprising that you would return true given a null set to match against.

Reply:
True. But an empty list that always returned false would be pretty useless here